plugins/catalog-backend/src/processing/DefaultCatalogProcessingEngine.ts

```diff
@@ -295,7 +295,7 @@ export class DefaultCatalogProcessingEngine {
                 });
               oldRelationSources = new Map(
                 previous.relations.map(r => [
-                  `${r.source_entity_ref}:${r.type}`,
+                  `${r.source_entity_ref}:${r.type}->${r.target_entity_ref}`,
```

Comment:
You know what - at this point, the Map with the unique keys really doesn't have a purpose anymore. The source-type-target triplet is the _entire relation_ basically. So I'd suggest using a map of source refs instead!

```ts
let oldRelationSources: Set<string>();
// ...
oldRelationSources = new Set(previous.relations.map(r => r.source_entity_ref));
}
```

and then the same thing with the new sources, making that a set too.

Then the refs that need stitching is `oldRelationSources.symmetricDifference(newRelationSources)`.

Reply:
@freben Thanks so much for the comments! However, I'm confused. The way the code worked before #17190 seems to be what you describe (that PR fixed issue #17063). If the type of the relation changed between old and new, then I believe the source would not appear in the symmetric difference between old and new and not get stitched according to your suggested code change above.

Am I misunderstanding something? Thanks!

Reply:
Ahh. I think you're quite right, I over simplified things. I'll make another review as soon as I get the chance